optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -275,12 +278,19 @@ def gaudi_llama_repeat_kv(
 
 #  FusedScaledDotProductAttention
 class ModuleFusedSDPA(torch.nn.Module):
-    def __init__(self, fusedSDPA):
+    def __init__(self, fusedSDPA, scale, attention_dropout, enable_recompute=False):
         super().__init__()
         self._hpu_kernel_fsdpa = fusedSDPA
-
-    def forward(self, query, key, value, attn_mask, dropout_p, is_casual, scale, softmax_mode):
-        return self._hpu_kernel_fsdpa.apply(query, key, value, attn_mask, dropout_p, is_casual, scale, softmax_mode)
+        self.scale = scale
+        self.attention_dropout = attention_dropout
+        self.enable_recompute = enable_recompute
+
+    def forward(self, query, key, value, attn_mask, dropout_p, is_causal, scale, fast_softmax_mode):
+        from habana_frameworks.torch.hpex.experimental.transformer_engine import FusedAttention
```

Comment:
Consider renaming FusedAttention to FusedAttentionTE or FusedAttentionFp8 to make it clear on why we need if/else loop below.

Reply:
Done